plugin/src/main/java/org/opensearch/ml/action/prediction/TransportPredictionTaskAction.java

```diff
@@ -139,7 +141,10 @@ public void onResponse(MLModel mlModel) {
                             }
                         }, e -> {
                             log.error("Failed to Validate Access for ModelId " + modelId, e);
-                            wrappedListener.onFailure(e);
+                            wrappedListener
+                                .onFailure(
+                                    new OpenSearchStatusException("Failed to Validate Access for ModelId " + modelId, RestStatus.FORBIDDEN)
```

Comment:
The error message means this is some internal error. Before validating access successfully, how can we know this is forbidden ?

Reply:
Good point, but I'm not sure if `INTERNAL_SERVER_ERROR` will be applicable here too. 

Here it's invoking [validateModelGroupAccess](https://github.com/opensearch-project/ml-commons/blob/8ae85a37d608739e57310bd7b9851c41d26bb06a/plugin/src/main/java/org/opensearch/ml/helper/ModelAccessControlHelper.java#L77-L140) method from where I think various kind of exception (Exception, MLResourceNotFoundException, MLValidationException) can be sent.

Do you have any suggestion?

Reply:
> various kind of exception (Exception, MLResourceNotFoundException, MLValidationException) can be sent.
> 
> Do you have any suggestion?

You could do a text search of the message for the exception name and/or if there is a "Status:" field in the message.

Reply:
@dbwiddis Could you please take a look at the current commit?